smart-contracts/contracts/cl-vault/src/helpers.rs

```diff
@@ -290,147 +326,201 @@ pub fn get_max_utilization_for_ratio(
     }
 }
 
-pub fn get_liquidity_amount_for_unused_funds(
-    deps: DepsMut,
-    env: &Env,
-    additional_excluded_funds: (Uint128, Uint128),
-) -> Result<Decimal256, ContractError> {
-    // first get the ratio of token0:token1 in the position.
-    let p = get_position(deps.storage, &deps.querier)?;
-    // if there is no position, then we can assume that there are 0 unused funds
-    if p.position.is_none() {
-        return Ok(Decimal256::zero());
+// TODO, allocating funds gives a ratio for each position, we should theb
+pub fn allocate_funds_per_position(
+    _deps: DepsMut,
+    positions: Vec<(Position, FullPositionBreakdown)>,
+    spot_price: Decimal,
+    token0: Uint128,
+    token1: Uint128,
+) -> Result<Vec<(Position, Uint128, Uint128)>, ContractError> {
+    // get the wanted ratio per position
+    let ps = get_min_ratio_per_position(positions, spot_price)?;
+
+    let (total0, total1) = ps.iter().try_fold(
```

Comment:
is this step required if we normalize the ratio two lines before? total0 and total1 should be 1 right?